### PR TITLE
DHFPROD-10422: BE: Wrong number of load records

### PR DESCRIPTION
**Pull request by prolonMarkLogic**
#### Checklist: 
```diff
- Note: do not change the below
```

- ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [n/a] Added Tests
- [n/a] Ran newly added/edited cypress tests on Firefox locally
  

- ##### Reviewer:

- [x] Reviewed Tests
- [n/a] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java [hunk 1]
```diff
@@ -505,7 +505,7 @@ private RunStepResponse runIngester(RunStepResponse runStepResponse, Collection<
                 listener.onStatusChange(runStepResponse.getJobId(), 100, stepStatus, stepMetrics.getSuccessfulEventsCount(), stepMetrics.getFailedEventsCount(), "Ingestion completed");
             });
 
-            runStepResponse.setCounts(stepMetrics.getSuccessfulEventsCount() + stepMetrics.getFailedEventsCount(),stepMetrics.getSuccessfulEventsCount(), stepMetrics.getFailedEventsCount(), stepMetrics.getSuccessfulBatchesCount(), stepMetrics.getFailedBatchesCount());
+            runStepResponse.setCounts(stepMetrics.getSuccessfulEventsCount() + stepMetrics.getFailedEventsCount(),count.get(), stepMetrics.getFailedEventsCount(), stepMetrics.getSuccessfulBatchesCount(), stepMetrics.getFailedBatchesCount());
```
Comment (aniamy) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java:508]: Shouldn't we have some testing in BE or FE to cover this?

Comment (prolonMarkLogic) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java:508]: Done, thanks!

**Comment by ngodugu-marklogic** [comment]
@prolonMarkLogic Load looks good but when I do Merge, for example in the attached video, the merge shows written documents are 1 but when I explore I see 4. Is that expected?

https://github.com/marklogic/marklogic-data-hub/assets/74939216/a5bf5a6e-31f5-4163-a2f1-56902c47b036

**Comment by prolonMarkLogic** [comment]
> @prolonMarkLogic Load looks good but when I do Merge, for example in the attached video, the merge shows written documents are 1 but when I explore I see 4. Is that expected?
> 
>  Screen.Recording.2023-08-11.at.9.47.42.AM.mov

This part will be covered on this ticket: DHFPROD-10567

**Review by ngodugu-marklogic — APPROVED** [review]

**Pull request merged**